File: frontend/src/types/incident.ts
export interface AIAnalysis {
  summary: string;
  root_cause: string;
  recommendation: string;
  severity: string;
}

export interface IncidentDetails {
  pod_name: string;
  namespace: string;
  status: string;
  phase: string;
  restart_count: number;
  node_name: string;
  container_name: string;
  container_image: string;
  image_pull_policy: string;
  container_state: string;
  last_termination_reason: string;
  resource_requests: Record<string, string>;
  resource_limits: Record<string, string>;
  liveness_probe: boolean;
  readiness_probe: boolean;
  logs: string;
  events: string[];
}

export interface IncidentHistory {
  incident: IncidentDetails;
  analysis: AIAnalysis;
}

File: backend/app/schemas/incident.py
from pydantic import BaseModel


class Incident(BaseModel):
    pod_name: str
    namespace: str

    status: str
    phase: str

    restart_count: int

    node_name: str

    container_name: str
    container_image: str

    image_pull_policy: str

    container_state: str

    last_termination_reason: str | None

    resource_requests: dict[str, str]
    resource_limits: dict[str, str]

    liveness_probe: bool
    readiness_probe: bool

    logs: str

    events: list[str]